eclipse-scout-core/src/desktop/outline/SearchOutline.ts

```diff
@@ -186,4 +270,133 @@ export class SearchOutline extends Outline implements SearchOutlineModel {
       elementToFocus.select();
     }
   }
+
+  updateSearchStates() {
+    const searchStates = new Set<SearchState>();
+
+    for (const page of this.nodes as SearchPage[]) {
+      if (page.searchState) {
+        searchStates.add(page.searchState);
+      }
+    }
+
+    this.setSearchStates(searchStates);
+  }
+
+  setSearchStates(searchStates: Set<SearchState>) {
+    searchStates ||= new Set();
+
+    const searchStatesToAdd = new Set<SearchState>();
+    for (const searchState of searchStates) {
+      if (this._searchStates.has(searchState)) {
+        continue;
+      }
+      searchStatesToAdd.add(searchState);
+    }
+
+    const searchStatesToRemove = new Set<SearchState>();
+    for (const searchState of this._searchStates) {
+      if (searchStates.has(searchState)) {
+        continue;
+      }
+      searchStatesToRemove.add(searchState);
+    }
+
+    searchStatesToAdd.forEach(searchState => this._addSearchState(searchState));
+    searchStatesToRemove.forEach(searchState => this._removeSearchState(searchState));
+  }
+
+  protected _addSearchState(searchState: SearchState) {
```

Comment:
I would integrate add and removeSearchState into setSearchStates so updateSearchStates is called only once. onSearchStateDestroy can call a public removeSearchState method that copies the existing states, remove the given state and calls setSearchStates()